server/player/chat/chat.go

```diff
@@ -15,12 +15,12 @@ var Global = New()
 // formatted messages to the chat.
 type Chat struct {
 	m           sync.Mutex
-	subscribers []Subscriber
+	subscribers map[Subscriber]struct{}
 }
 
-// New returns a new chat. A zero value is, however, usually sufficient for use.
```

Comment:
Good thing you removed this. This wasn't actually accurate back then, because the `sync.Mutex` would be copied which would break things.